File: esthesis-core-ui/src/app/shared/services/dto/oidc-config-dto.ts
export interface OidcConfigDto {
  authority: string;
  redirectUrl: string;
  postLogoutRedirectUri: string;
  clientId: string;
  scope: string;
  responseType: string;
  silentRenew: boolean;
  useRefreshToken: boolean;
  renewTimeBeforeTokenExpiresInSeconds: number;
  maxIdTokenIatOffsetAllowedInSeconds: number;
  ignoreNonceAfterRefresh: boolean;
  secureRoutes: string[];
}

File: esthesis-core-backend/services/srv-public-access/srv-public-access-client/src/main/java/esthesis/service/publicaccess/dto/OidcConfigDTO.java
package esthesis.service.publicaccess.dto;

import lombok.Data;

/**
 * A representation of the OIDC configuration.
 */
@Data
public class OidcConfigDTO {

	private String authority;
	private String redirectUrl;
	private String postLogoutRedirectUri;
	private String clientId;
	private String scope;
	private String responseType;
	private boolean silentRenew;
	private boolean useRefreshToken;
	private int renewTimeBeforeTokenExpiresInSeconds;
	private int maxIdTokenIatOffsetAllowedInSeconds;
	private boolean ignoreNonceAfterRefresh;
	private String[] secureRoutes;
}
